web/pdf_print_service.js

```diff
@@ -200,7 +211,17 @@
       if (!activeService) {
         console.error('Expected print service to be initialized.');
       }
-      activeService.renderPages().then(startPrint, abort);
+      var activeServiceOnEntry = activeService;
+      activeService.renderPages().then(function() {
+        activeServiceOnEntry.throwIfInactive();
+        return startPrint(activeServiceOnEntry);
+      }).then(null, function() {
```

Comment:
Nit: I may very well be overlooking something here, but couldn't you just use `catch` here instead of these two `then`s? I.e. the code below simply becomes:
```javascript
}).catch(function() {
  if (activeServiceOnEntry.isActive()) {
    abort();
  }
});
```